tests/Aspire.Hosting.Oracle.Tests/OracleFunctionalTests.cs

```diff
@@ -0,0 +1,392 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Components.Common.Tests;
+using Aspire.Hosting.Utils;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using Microsoft.EntityFrameworkCore.Storage;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Polly;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Aspire.Hosting.Oracle.Tests;
+
+public class OracleFunctionalTests(ITestOutputHelper testOutputHelper)
+{
+    [Fact]
+    [RequiresDocker]
+    public async Task VerifyOracleResource()
+    {
+        var cts = new CancellationTokenSource(TimeSpan.FromMinutes(3));
+        var pipeline = new ResiliencePipelineBuilder()
+            .AddRetry(new() { MaxRetryAttempts = 10, Delay = TimeSpan.FromSeconds(2) })
+            .Build();
+
+        var builder = CreateDistributedApplicationBuilder();
+
+        var oracleDbName = "freepdb1";
+
+        var oracle = builder.AddOracle("oracle");
+        var db = oracle.AddDatabase(oracleDbName);
+
+        using var app = builder.Build();
+
+        await app.StartAsync();
+
+        var hb = Host.CreateApplicationBuilder();
+
+        hb.Configuration.AddInMemoryCollection(new Dictionary<string, string?>
+        {
+            [$"ConnectionStrings:{db.Resource.Name}"] = await db.Resource.ConnectionStringExpression.GetValueAsync(default)
+        });
+
+        hb.AddOracleDatabaseDbContext<TestDbContext>(db.Resource.Name);
+
+        using var host = hb.Build();
+
+        await host.StartAsync();
+
+        // Wait until the database is available
+        await pipeline.ExecuteAsync(async token =>
+        {
+            var dbContext = host.Services.GetRequiredService<TestDbContext>();
+            var databaseCreator = (RelationalDatabaseCreator)dbContext.Database.GetService<IDatabaseCreator>();
+            Assert.True(await databaseCreator.CanConnectAsync(token));
+        }, cts.Token);
+
+        await pipeline.ExecuteAsync(async token =>
```

Comment:
Could you help me understand why are retries needed here? Shouldn't `await connection.OpenAsync(token);` always pass now, because of the block above that checked for `CanConnectAsync`?

Reply:
Nothing says if `CanConnectAsync()` can return `true` and yet the database not be ready to execute actual queries. Not sure about how it's implemented for each provider. Just a safety since we already have a lot of instability with DB tests.

But we could also just say that the previous call is sufficient for this test and remove this call altogether.

Reply:
Safety sounds good to me! 👍